### PR TITLE
fix: support receipt/record queries with nonce

### LINKED ISSUE #12652: (mod) Support receipt and record queries for child transactions
### Problem

We want to be able to query for a specific child receipt or record by nonce, instead of needing to query the top-level `TransactionID` with e.g. `include_child_receipts=true`.

### Solution

Look up child receipts and records from the `RecordCache` using their top-level `TransactionID`; then check the list of child records for a matching `nonce`.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #12652 
 - Fix `DeduplicationCache` comparator to include `nonce`.
 - Remove some misleading comments in `RecordCacheImpl`.
 - Update receipt and record queries to fetch a specific child by `TransactionID` nonce.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
Makes sense to me. Thanks @tinker-michaelj

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/recordcache/RecordCacheImpl.java [hunk 1]
```diff
@@ -311,6 +303,11 @@ private void removeExpiredTransactions(
     @Nullable
     @Override
     public History getHistory(@NonNull TransactionID transactionID) {
+        logger.info(
+                "Getting history for transaction ID: {} (contained in dedup cache? {}) - history = {}",
+                transactionID,
+                deduplicationCache.contains(transactionID),
+                histories.get(transactionID));
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/recordcache/RecordCacheImpl.java:310]: This can be deleted

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @tinker-michaelj

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java [hunk 2]
```diff
@@ -84,23 +84,40 @@ public Response findResponse(@NonNull final QueryContext context, @NonNull final
...
+                    final var maybeRecord = history.childRecords().stream()
+                            .filter(record -> transactionId.equals(record.transactionID()))
+                            .findFirst();
+                    if (maybeRecord.isEmpty()) {
+                        responseBuilder.header(header.copyBuilder()
+                                .nodeTransactionPrecheckCode(RECEIPT_NOT_FOUND)
+                                .build());
+                    } else {
+                        responseBuilder.receipt(maybeRecord.get().receipt());
+                    }
```
Comment (mhess-swl) [inline hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java:120]: Seems worth extracting to a common method, but definitely not a PR blocker

**Review by mhess-swl — APPROVED** [review]
LGTM, thanks @tinker-michaelj!

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java:120]: True Matt, noticed the duplicated pattern as well...then thought these queries aren't likely to change any time soon, and decided to leave it for now. 🤷

**Pull request merged**